### PR TITLE
Heated Chamber for Graphical LCD

### PR DESCRIPTION
**Pull request by InsanityAutomation**
This adds an icon for heated chambers based off of the work here : https://www.thingiverse.com/thing:3600505

Ive added the control around icon animation, but still need to decide on how to set MAX_HOTEND_DRAW with this. Should this take priority over a second hotend or should it be configurable in some way?

Figured id get other opinions before wrapping up the last bit.

### CODE REVIEW

File: Marlin/src/lcd/dogm/status_screen_DOGM.cpp [hunk 1]
```diff
@@ -366,6 +406,10 @@ void MarlinUI::draw_status_screen() {
       _draw_heater_status(-1, blink);
     #endif
 
+    #if HAS_HEATED_CHAMBER
+	    _draw_chamber_status(blink);
+	  #endif
```
Comment (thinkyhead) [inline Marlin/src/lcd/dogm/status_screen_DOGM.cpp]: tabs fell in! 😉

**Comment by InsanityAutomation** [comment]
Whoops! Didn't notice the tabs in the stuff that I copied from for a starting point....

File: Marlin/src/lcd/dogm/status_screen_DOGM.cpp (hunk 1 above)
Comment (tpruvot) [reply Marlin/src/lcd/dogm/status_screen_DOGM.cpp]: image

**Comment by alexscarbro** [comment]
Thank you for adding the chamber temperature display functionality. I'm currently building this in to a controller with solid state relays for the chamber heater & fan and its very useful.

Unfortunately, When building with Arduino GUI on Windows 10, the _draw_chamber_status() function definition triggers a SegFault in the compiler. When building in PlatformIO/Atom, it complains that the function definition isn't expected at its current location. 

If it's moved out of the _draw_heater_status() function definition, (above in my case) it compiles correctly.

Any ideas as to why?

Thanks,
Alex

**Comment by InsanityAutomation** [comment]
I'm normally working with platformio and vscode, atom isn't really supported anymore. I'll check the Arduino ide and see what's up.

**Comment by thinkyhead** [comment]
@alexscarbro — Try removing `FORCE_INLINE` from the front of `_draw_chamber_status` and see if it helps.

**Comment by alexscarbro** [comment]
Hello,

Yes, I seem to recall I tried that, but to no avail. Instead I moved it out of the enclosing function and all was well.

Best regards,
Alex

Sent from my iPhone

On 11 Jul 2019, at 20:35, Scott Lahteine <notifications@github.com<mailto:notifications@github.com>> wrote:


@alexscarbro<https://github.com/alexscarbro> — Try removing FORCE_INLINE from the front of _draw_chamber_status and see if it helps.

**Comment by thinkyhead** [comment]
Oh, geez. Well that is a typo or an error. We don't want to define functions inside other functions. I will post a patch shortly.

**Pull request merged**